macros/cross_db_utils/dateadd.sql

```diff
@@ -16,8 +16,8 @@
 
 {% macro bigquery__dateadd(datepart, interval, from_date_or_timestamp) %}
 
-    date_add(
-        cast( {{ from_date_or_timestamp }} as date),
+    datetime_add(
```

Comment:
I think this would be better if we cast to a `timestamp` instead of a `datetime`, and then use `timestamp_add` accordingly. Timestamps have timezones, whereas datetimes do not. More info on that one [here](https://stackoverflow.com/questions/47713945/bigquery-datetime-vs-timestamp).

Reply:
so when i cast it to `timestamp` and use `timestamp_add` i get:
`TIMESTAMP_ADD does not support the MONTH date part` ?? haven't dug too deep as to why, but just letting you know that's the error i get